CommandLine/emake-tests/Parsing/parser-test.cpp

```diff
@@ -40,6 +40,56 @@ void assert_identifier_is(AST::Node *node, std::string_view name) {
 }
 
 TEST(ParserTest, Basics) {
+  ParserTester test{"(x ? y : z ? a : (z[5](6)));"};
+
+  auto node = test->TryParseStatement();
+  ASSERT_EQ(node->type, AST::NodeType::PARENTHETICAL);
+
+  auto *expr = node->As<AST::Parenthetical>()->expression.get();
+  ASSERT_EQ(expr->type, AST::NodeType::TERNARY_EXPRESSION);
+
+  auto *ternary = expr->As<AST::TernaryExpression>();
+  auto *cond = ternary->condition.get();
+  auto *true_ = ternary->true_expression.get();
+  auto *false_ = ternary->false_expression.get();
+  assert_identifier_is(cond, "x");
+  assert_identifier_is(true_, "y");
+
+  ASSERT_EQ(false_->type, AST::NodeType::TERNARY_EXPRESSION);
+
+  ternary = false_->As<AST::TernaryExpression>();
+
+  cond = ternary->condition.get();
+  true_ = ternary->true_expression.get();
```

Comment:
nit: Usually variables ending in `_` indicate private fields, so you might want to prefer a name like `true_exp`. 

As another odd point to note: names containing `__` (two underscores) or starting in an underscore *and* capital letter are reserved for the compiler/standard libraries.